Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +212,49 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
+        $document = $this->dm->find(null, $path);
 
-        return $this->dm->find(null, $path);
+        if ($document instanceof AutoRouteInterface) {
+            return $document;
+        }
+
+        return null;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * Convert the given generic document to an auto route document, migrating any children which it may have.
+     *
+     * @param Generic $document
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function migrateGenericToAutoRoute(Generic $document, $contentDocument, $autoRouteTag, $routeType)
+    {
+        $autoRouteClassName = $this->autoRouteFqcn;
```

Comment:
> maybe we should have a method to migrate the class in PHPCR ODM ..

we have DocumentMigrateClassCommand - the logic should be refactored
into a helper class that the command simply calls, instead of coding the
logic in the command directly.